src/inotify/mod.rs

```diff
@@ -116,73 +192,60 @@ fn handle_event(event: wrapper::Event, tx: &Sender<Event>, paths: &Arc<RwLock<Ha
 
 impl Watcher for INotifyWatcher {
   fn new(tx: Sender<Event>) -> Result<INotifyWatcher, Error> {
-    let mut it = match INotify::init() {
-      Ok(i) => INotifyWatcher {
-        inotify: i,
-        tx: tx,
-        watches: HashMap::new(), // TODO: use bimap?
-        paths: Arc::new(RwLock::new(HashMap::new()))
-      },
-      Err(e) => return Err(Error::Io(e))
-    };
+    INotify::init()
+      .and_then(|inotify| EventLoop::new().map(|l| (inotify, l)))
+      .and_then(|(inotify, mut event_loop)| {
+        let inotify_fd = inotify.fd;
+        let evented_inotify = mio::unix::EventedFd(&inotify_fd);
+
+        let handler = INotifyHandler {
+          inotify: inotify,
+          tx: tx,
+          watches: HashMap::new(), // TODO: use bimap?
```

Comment:
Can you remove this TODO? I realise it is in the origin code, but it hasn't been relevant is a while, so might as well clean it up...